tests/webSocket-spec.coffee

```diff
@@ -0,0 +1,523 @@
+CallbacksBuffer     = require '../src/callbacks-buffer'
+
+options =
+  reconnectionMaxTimeout: 30e3
+  heartbeatTimeout: 3e3
+  heartbeatInterval: 5e3
+  reconnect: true
+  onDisconnect: 'buffer'
+
+describe 'webSocket', ->
+
+  describe 'initialization', ->
+
+    it 'should construct URL that matches Really URL scheme when domain is passed', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      expect(ws.url).toEqual "wss://a6bcc.api.really.io/v#{protocol.clientVersion}/socket"
+
+    it 'should initialize socket', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      expect(ws.socket).toBeNull()
+
+    it 'should initialize callbacks buffer', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      expect(ws.callbacksBuffer).toEqual new CallbacksBuffer()
+
+    it 'should initialize messages buffer', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      expect(ws._messagesBuffer).toEqual []
+
+    it 'should set initialized to false', ->
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options)
+      expect(ws.initialized).toBeFalsy()
+
+    it 'should put appropriate default values for options if not supplied', ->
+      defaultValues =
+        reconnectionMaxTimeout: 30e3
+        heartbeatTimeout: 2e3
+        heartbeatInterval: 5e3
+        reconnect: true
+        onDisconnect: 'buffer'
+
+      options0 = {}
+      ws = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options0)
+      expect(ws.options).toEqual defaultValues
+
+      options1 =
+        reconnectionMaxTimeout: 25e3
+
+      ws1 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options1)
+      expect(ws1.options).toEqual options1
+
+      options2 =
+        reconnectionMaxTimeout: 25e3
+        heartbeatTimeout: 4e3
+      
+      ws2 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options2)
+      expect(ws2.options).toEqual options2
+
+      options3 =
+        reconnectionMaxTimeout: 25e3
+        heartbeatTimeout: 4e3
+        heartbeatInterval: 4e3
+      
+      ws3 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options3)
+      expect(ws3.options).toEqual options3
+
+      options4 =
+        reconnectionMaxTimeout: 25e3
+        heartbeatTimeout: 4e3
+        heartbeatInterval: 4e3
+        reconnect: false
+      
+      ws4 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options4)
+      expect(ws4.options).toEqual options4
+
+      options5 =
+        reconnectionMaxTimeout: 25e3
+        heartbeatTimeout: 4e3
+        heartbeatInterval: 4e3
+        reconnect: false
+        onDisconnect: 'final'
+
+      ws5 = new WebSocketTransport('wss://a6bcc.api.really.io', 'ibj88w5aye', options5)
+      expect(ws5.options).toEqual options5
```

Comment:
IMHO this is over; you are testing if `_.defaults` really works in different cases which is already tested in lodash, it's better if you test that `_.defaults` is called with the proper defaults and the passed options